src/main/java/ch/njol/skript/conditions/CondContains.java

```diff
@@ -34,146 +34,131 @@
 import ch.njol.util.Checker;
 import ch.njol.util.Kleenean;
 import ch.njol.util.StringUtils;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Contains")
-@Description("Checks whether an inventory contains the given item, a text contains another piece of text, or a list of objects (e.g. a {list variable::*}) contains another object.")
+@Description("Checks whether an inventory contains an item, a text contains another piece of text, " +
+		"or a list (e.g. {list variable::*} or 'drops') contains another object.")
 @Examples({"block contains 20 cobblestone",
-		"player has 4 flint and 2 iron ingots"})
+		"player has 4 flint and 2 iron ingots",
+		"{list::*} contains 5"})
 @Since("1.0")
 public class CondContains extends Condition {
-	
+
 	static {
 		Skript.registerCondition(CondContains.class,
-				"%inventories% ha(s|ve) %itemtypes% [in [(the[ir]|his|her|its)] inventory]",
+				"%inventories% (has|have) %itemtypes% [in [(the[ir]|his|her|its)] inventory]",
+				"%inventories% (doesn't|does not|do not|don't) have %itemtypes% [in [(the[ir]|his|her|its)] inventory]",
 				"%inventories/strings/objects% contain[s] %itemtypes/strings/objects%",
-				"%inventories% do[es](n't| not) have %itemtypes% [in [(the[ir]|his|her|its)] inventory]",
-				"%inventories/strings/objects% do[es](n't| not) contain %itemtypes/strings/objects%");
+				"%inventories/strings/objects% (doesn't|does not|do not|don't) contain %itemtypes/strings/objects%",
+				"[the] list [of] %objects% (doesn't|does not|do not|don't) contain %objects%", // comes before 'contains' because of a parser bug
+				"[the] list [of] %objects% contain[s] %objects%",
+				"(all|1¦any|2¦none) of %strings% contain[s] %strings%");
 	}
-	
+
 	@SuppressWarnings("null")
 	private Expression<?> containers;
 	@SuppressWarnings("null")
 	private Expression<?> items;
-	
-	@SuppressWarnings({"unchecked", "null", "unused"})
+	private boolean isListCheck, isStringCheck, isAnyNoneOf;
+
+	@SuppressWarnings({"null", "unchecked"})
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
-		containers = exprs[0].getConvertedExpression(Object.class);
-		if (containers == null)
-			return false;
-		if (!(containers instanceof Variable) && !String.class.isAssignableFrom(containers.getReturnType()) && !Inventory.class.isAssignableFrom(containers.getReturnType())
-				&& !containers.getReturnType().equals(Object.class)) {
-			final ParseLogHandler h = SkriptLogger.startParseLogHandler();
-			try {
-				Expression<?> c = containers.getConvertedExpression(String.class);
-				if (c == null)
-					c = containers.getConvertedExpression(Inventory.class);
-				if (c == null) {
-					h.printError();
-					return false;
-				}
-				containers = c;
-				h.printLog();
-			} finally {
-				h.stop();
-			}
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
```

Comment:
You deleted a local log handler. Are you sure it causes no adverse effects?

Reply:
I made it so it will try to convert the expression but don't do anything if it couldn't, so no need to error logs.